hedera-mirror-rosetta/app/persistence/transaction/transaction_test.go

```diff
@@ -0,0 +1,738 @@
+	}
+	dbTransactionTypes = []transactionType{
+		{
+			ProtoID: 12,
+			Name:    "CRYPTODELETE",
+		},
+		{
+			ProtoID: 14,
+			Name:    "CRYPTOTRANSFER",
+		},
+	}
+	dbTransactionResults = []transactionResult{
+		{ProtoID: 0, Result: "OK"},
+		{ProtoID: 1, Result: "INVALID_TRANSACTION"},
+	}
+	expectedTransaction = &types.Transaction{
+		Hash:       hexutils.SafeAddHexPrefix(hex.EncodeToString(hash)),
+		Operations: operations,
+	}
+	operations = []*types.Operation{
+		{
+			Index:   0,
+			Type:    "CRYPTOTRANSFER",
+			Status:  "OK",
+			Account: firstAccount,
+			Amount:  &amount,
+		},
+		{
+			Index:   1,
+			Type:    "CRYPTODELETE",
+			Status:  "OK",
+			Account: secondAccount,
+			Amount:  &amount,
+		},
+	}
+	tRepoTypes = map[int]string{
+		12: "CRYPTODELETE",
+		14: "CRYPTOTRANSFER",
+	}
+	tRepoStatuses = map[int]string{
+		0: "OK",
+		1: "INVALID_TRANSACTION",
+	}
+	tRepoTypesAsArray = []string{"CRYPTODELETE", "CRYPTOTRANSFER"}
+)
+
+func TestShouldSuccessFindBetween(t *testing.T) {
+	// given
+	tr, mock := setupRepository(t)
+	defer tr.dbClient.DB().Close()
+
+	mock.ExpectQuery(regexp.QuoteMeta(whereClauseBetweenConsensus)).
+		WithArgs(int64(1), int64(2)).
+		WillReturnRows(willReturnRows(transactionColumns, dbTransactions))
+	mock.ExpectQuery(regexp.QuoteMeta(whereTimestampsInConsensusTimestamp)).
+		WithArgs("1,2").
+		WillReturnRows(willReturnRows(cryptoTransferColumns, dbCryptoTransfers))
+	mock.ExpectQuery(regexp.QuoteMeta(selectTransactionTypes)).
+		WillReturnRows(willReturnRows(transactionTypeColumns, dbTransactionTypes))
+	mock.ExpectQuery(regexp.QuoteMeta(selectTransactionResults)).
+		WillReturnRows(willReturnRows(transactionResultColumns, dbTransactionResults))
+
+	// when
+	result, err := tr.FindBetween(1, 2)
+
+	// then
+	assert.NoError(t, mock.ExpectationsWereMet())
+	assert.Nil(t, err)
+	assert.IsType(t, []*types.Transaction{expectedTransaction}, result)
+}
+
+func TestShouldFailFindBetweenNoTypes(t *testing.T) {
+	// given
+	tr, mock := setupRepository(t)
+	defer tr.dbClient.DB().Close()
+
+	mock.ExpectQuery(regexp.QuoteMeta(whereClauseBetweenConsensus)).
+		WithArgs(int64(1), int64(2)).
+		WillReturnRows(willReturnRows(transactionColumns, dbTransactions))
+	mock.ExpectQuery(regexp.QuoteMeta(whereTimestampsInConsensusTimestamp)).
+		WithArgs("1,2").
+		WillReturnRows(willReturnRows(cryptoTransferColumns, dbCryptoTransfers))
+	mock.ExpectQuery(regexp.QuoteMeta(selectTransactionTypes)).
+		WillReturnRows(willReturnRows(transactionTypeColumns))
+
+	// when
+	result, err := tr.FindBetween(1, 2)
+
+	// then
+	assert.NoError(t, mock.ExpectationsWereMet())
+	assert.Nil(t, result)
+	assert.IsType(t, rTypes.Error{}, *err)
+}
+
+func TestShouldFailFindBetweenEndBeforeStart(t *testing.T) {
+	// given
+	tr, _ := setupRepository(t)
+	defer tr.dbClient.DB().Close()
+
+	// when
+	result, err := tr.FindBetween(2, 1)
+
+	// then
+	assert.Nil(t, result)
+	assert.IsType(t, rTypes.Error{}, *err)
```

Comment:
IMO you should assert the concrete error that was returned not only that it was error